index.js

```diff
@@ -35,4 +37,37 @@ fetch(FORUM_API)
     isLoading = false;
   });
 
+const displayTopicList = (topicList) => {
+  for (const topic of topicList) {
+    const categoryId = topic["category_id"];
+    if (!(categoryId in forumCategoriesObj)) {
+      continue;
+    }
+    const category = forumCategoriesObj[categoryId].className;
+    const categoryText = forumCategoriesObj[categoryId].category;
+    displayTopic(topic, category, categoryText);
+  }
+};
+
+const displayTopic = (topic, category, categoryText) => {
+  let post = `<tr> 
+    <td>
+      <span id='post-title'>
+        <a href='${FORUM_TOPIC}/${topic.slug}' target='_blank'>
+          ${topic.title}
+        </a>
+      </span>
+      <div id='post-category'>
+        <a class='${category}' href='${FORUM_CATEGORY}/${category}' target='_blank'>
+          ${categoryText}
+        </a>
+      </div>
+    </td>
+    <td></td>
+    <td></td>
+    <td></td>
+  </tr>`;
+  postsContainer.innerHTML += post;
+};
```

Comment:
After you change the categories map name and its member property name,
you don't really have to store its text and className in other variables to make
clear what it is as the property names become self explanatory enough

```suggestion
const displayTopic = (topic) => {
  const category = supportedTopicCategories[topic["category_id"]]
  let post = `<tr> 
    <td>
      <span id='post-title'>
        <a href='${FORUM_TOPIC}/${topic.slug}' target='_blank'>
          ${topic.title}
        </a>
      </span>
      <div id='post-category'>
        <a class='${category.className}' href='${FORUM_CATEGORY}/${category.name}' target='_blank'>
          ${category.name}
        </a>
      </div>
    </td>
    <td></td>
    <td></td>
    <td></td>
  </tr>`;
  postsContainer.innerHTML += post;
};
```

Reply:
I see. Thanks!

Reply:
I agree, the code is much more readable now, thank you. 
 One more thing, since `className` property is used in both the `class` and `href` attribute, should we modify it to something else ?

Reply:
oh I see.. in that case we could change `className` to `name` and `name` to `title`, `longName`, `titleName`,... I'll leave that for you to decide